apps/meteor/app/livechat/server/lib/Livechat.js

```diff
@@ -638,13 +639,32 @@ export const Livechat = {
 		const extraQuery = await callbacks.run('livechat.applyRoomRestrictions', {});
 		const cursor = LivechatRooms.findByVisitorToken(token, extraQuery);
 		for await (const room of cursor) {
-			await FileUpload.removeFilesByRoomId(room._id);
-			await Messages.removeByRoomId(room._id);
+			const result = await Promise.allSettled([
+				FileUpload.removeFilesByRoomId(room._id),
+				Messages.removeByRoomId(room._id),
+				ReadReceipts.removeByRoomId(room._id),
+			]);
+
+			for (const r of result) {
```

Comment:
instead of looping through all promises, you could just use `Promise.all` and you'll get the error if any of them fails